SharkCage/CageManager/CageLabeler.h

```diff
@@ -0,0 +1,208 @@
+#include <Windows.h>
+#include <windowsx.h>
+#include <algorithm> 
+
+using namespace std;
+
+#include <objidl.h>
+#include <gdiplus.h>
+
+using namespace Gdiplus;
+#pragma comment (lib, "Gdiplus.lib")
+
+VOID DisplayTokenInCageWindow(HWND *hwnd);
+int GetBottomFromMonitor();
+
+HWND gotodesk_button;
+HHOOK hHook;
+
+class CageLabeler
+{
+public:
+	CageLabeler();
+	~CageLabeler();
+	bool Init();
+private:
+	bool CageLabeler::ShowCageWindow();
+	VOID CageLabeler::InitGdipPlisLib();
+};
+
+CageLabeler::CageLabeler()
+{
+	InitGdipPlisLib();
+}
+
+CageLabeler::~CageLabeler()
+{
+}
+
+VOID CageLabeler::InitGdipPlisLib()
+{
+	GdiplusStartupInput gdiplusStartupInput;
+	ULONG_PTR gdiplusToken;
+	GdiplusStartup(&gdiplusToken, &gdiplusStartupInput, NULL);
+}
+
+LRESULT CALLBACK WndProc(HWND hwnd, UINT msg, WPARAM w_param, LPARAM l_param)
+{
+	std::cout << msg << std::endl;
+	switch (msg)
+	{
+	case WM_CREATE:
+	{
+		gotodesk_button = CreateWindowEx(
+			NULL,
+			L"BUTTON",
+			L"Exit",
+			WS_TABSTOP | WS_VISIBLE | WS_CHILD | BS_DEFPUSHBUTTON,
+			10,
+			900,
+			100,
+			100,
+			hwnd,
+			NULL,
+			(HINSTANCE)GetWindowLong(hwnd, GWL_HINSTANCE),
+			NULL);
+
+		if (gotodesk_button != NULL)
+		{
+			SendMessage(gotodesk_button, BM_SETIMAGE, NULL, NULL);
+		}
+		else
+		{
+			std::wcout << L"Failed to create logout button Err " << GetLastError() << std::endl;
+		}
+		break;
+	}
+	case WM_COMMAND:
+	{
+		if (l_param == (LPARAM)gotodesk_button) {
+			// Close token
+		}
+		else {
+			break;
+		}
+	}
+	case WM_CLOSE:
+		std::cout << "Close window" << std::endl;
+		DestroyWindow(hwnd);
+		break;
+	case WM_DESTROY:
+		PostQuitMessage(0);
+		break;
+	case WM_PAINT:
+	{
```

Comment:
You can only create new variables in a scope. Without brackets, there is no new scope for the case of the switch. But also need not in every case a scope...Should we create for the readability scopes in each case? 

The break is not there because we have to consume the msg using the default window procedure in this case.

Reply:
* I'd go with consistency and use brackets everywhere
* this is dangerous, inserting another case between this one and the default case or changing the default case introduces a major bug! just call the default window procedure explicitely inside this case